src/rgw/rgw_bucket.cc

```diff
@@ -778,25 +778,38 @@ static void set_err_msg(std::string *sink, std::string msg)
     *sink = msg;
 }
 
-int RGWBucket::init(RGWRados *storage, RGWBucketAdminOpState& op_state)
+int RGWBucket::init(RGWRados *storage, RGWBucketAdminOpState& op_state,
+	std::string *err_msg)
 {
-  if (!storage)
+  std::string bucket_tenant;
+  if (!storage) {
+    set_err_msg(err_msg, "no storage!");
     return -EINVAL;
+  }
 
   store = storage;
 
   rgw_user user_id = op_state.get_user_id();
   tenant = user_id.tenant;
+  bucket_tenant = tenant;
```

Comment:
'bucket_tenant' assignment here will pass the new user's tenant to _RGW_Rados::get_bucket_info()_ call in line 796, thus failing with _ret code -2_ in case of original user being untenanted. That's because the 'bucket_tenant' in case of untenanted user will not be updated in line 791 due to absence of tenant/name splitting. This line can be removed safely.

Reply:
Vikhyat pointed out the '/' prefix I missed before the bucket name in the command. Will update the docs.